lib/matplotlib/axes/_axes.py

```diff
@@ -5425,7 +5378,11 @@ def table(self, **kwargs):
 
         %(Table)s
         """
-        return mtable.table(self, **kwargs)
+        return mtable.RegularTableCreator(self, **kwargs)
```

Comment:
This changes the return type of this function which it a API breaking change.

It also means that this no longer returns an `Artist` which is less than great.